test/runtests.py

```diff
@@ -153,6 +153,21 @@ def test_proxy_from_dhcp_wpad(self):
             fake_proxy_1.proc.kill()
             static_server.proc.kill()
 
+    def test_get_dns_wpad_urls_hostname_only(self):
+        from pac4cli.wpad import WPAD
+        wpad = WPAD(None, None)
+        dns_urls = wpad.get_dns_wpad_urls(["hostname.example.com"])
+        self.assertEqual(len(dns_urls), 1)
+        self.assertEqual(dns_urls[0], "http://wpad.example.com/wpad.dat")
+
+    def test_get_dns_wpad_urls_hostname_and_subdomain(self):
+        from pac4cli.wpad import WPAD
+        wpad = WPAD(None, None)
+        dns_urls = wpad.get_dns_wpad_urls(["hostname.subdomain.example.com"])
+        self.assertEqual(len(dns_urls), 2)
+        self.assertEqual(dns_urls[0], "http://wpad.subdomain.example.com/wpad.dat")
+        self.assertEqual(dns_urls[1], "http://wpad.example.com/wpad.dat")
```

Comment:
Because the order in which `get_dns_wpad_urls` returns the results should probably be considered an implementation detail, maybe we can avoid testing that. E.g.
```
self.assertEqual(sorted(dns_urls), sorted([
    "http://wpad.example.com/wpad.dat", 
    "http://wpad.subdomain.example.com/wpad.dat",
]))
```
(Also, should decide whether `http://wpad.hostname.subdomain.example.com` should be part of the result, and either update this test+the code, or update the code comment elsewhere.)

Reply:
Following the wpad DNS discovery protocol, the order is significant. That is, the order is part of the promise not an implementation details (per my understanding of the protocol of course)

Reply:
Also given a `hostname.subdomain.example. com`, this is the fqdn of the machine at hand. Meaning the `hostname` part refers to the local hostname of the machine. Hence, we should not trt the `http://wpad.hostname.subdomain.example.com/wpad.dat`. But to be supper sure, I'll consult the NetworkManager documentation to make sure that the domains provided here are the indeed the domain names assigned to the machine in the network it is connected to (fqdn of the host)

Reply:
Good point!